glean-core/src/upload/mod.rs

```diff
@@ -112,7 +117,16 @@ pub enum PingUploadTask {
     /// A flag signaling that the pending pings directories are not done being processed,
     /// thus the requester should wait and come back later.
     Wait,
-    /// A flag signaling that the pending pings queue is empty and requester is done.
+    /// A flag signaling that requester doesn't need to request any more upload tasks at this moment.
+    ///
+    /// There are two possibilities for this scenario:
+    /// * Pending pings queue is empty, no more pings to request;
+    /// * Requester has reported more than MAX_RECOVERABLE_FAILURES_PER_PERIOD
+    ///   recoverable upload failures on the same period[1]
+    ///   and should stop requesting at this moment.
+    ///
+    /// [1]: A "period" starts when a requester gets a new `PingUploadTask::Upload(PingRequest)`
```

Comment:
Not really happy with the "period" terminology here, couldn't think of anything better at the time. Please help me in bikeshedding.

Reply:
Time window? Window? Time-frame?